src/Chip/index.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Icon from '../Icon';
+
+const TAG_VARIANTS = [
+  'button',
+  'a',
+];
+
+const STYLE_VARIANTS = [
+  'light',
+  'dark',
+];
+
+const Chip = ({
+  as,
+  children,
+  className,
+  variant,
+  active,
+  iconBefore,
+  iconAfter,
+  ...props
+}) => {
+  const elementType = TAG_VARIANTS.includes(as) ? as : 'button';
+  const elementProps = {
+    ...props,
+    className: classNames(
+      'pgn__chip',
+      `pgn__chip-${variant}`,
+      { active },
+      className,
+    ),
+  };
+  const elementContent = (
+    <>
+      {iconBefore && <Icon className={classNames('chip-icon-before')} src={iconBefore} />}
+      {children}
+      {iconAfter && <Icon className={classNames('chip-icon-after')} src={iconAfter} />}
+    </>
+  );
+  return React.createElement(props.href ? 'a' : elementType, elementProps, elementContent);
+};
+
+Chip.propTypes = {
+  /** Specifies the base HTML element. */
+  as: PropTypes.elementType,
```

Comment:
In the component implementation you only allow 'button' or 'a' elements, maybe you can define this props as `PropTypes.oneOf(['button', 'a'])` or something like this? This way you don't need to check if the prop is correct like you do [here](https://github.com/edx/paragon/pull/911/files#diff-b7460be0253af56f24ad7fb65320ffdac22f6f0f26b87ff93ca36943c3cac36dR26)

I'm also not sure if we need to restrict the element to only 'a' or 'button', I think everywhere else in Paragon `as` is always an arbitrary element, maybe we should do the same here?

Reply:
According to the task it must either `button` or `a`. But I agree, everywhere it is arbitrary and here would be good to have the same behavior. @adamstankiewicz what do you think about this?

Reply:
Ohh I see, I missed that in task description, sorry.

Let's see what Adam thinks then 🙂

Reply:
Yes, I agree. If we can build this to keep `as` consistent with how it's used every else, we should do that :)